docs/pages/kubernetes-access/guides/discovery.mdx

```diff
@@ -0,0 +1,86 @@
+---
+title: Kubernetes Discovery Guides (Preview)
+description: Detailed guides for configuring Kubernetes Auto-discovery.
+layout: tocless-doc
+---
+
+## Discovery
+
+Kubernetes cluster auto-discovery is a Teleport feature that allows the Teleport process automatically discover and enroll Kubernetes clusters from different cloud providers. 
+The discovery process does not install any component on the cluster but requires Teleport to have access to the cluster API and minimal permissions.
+
+The discovery consists of two distinct steps:
+
+### Discovery
+
+The discovery process is responsible for scanning the cloud providers and checking if any Kubernetes cluster match the desired filtering labels. 
+When a new cluster is discovered, it registers that cluster in Teleport together with the cloud access parameters.
+
+This process is also responsible for removing the cluster when it has been deleted or when it no longer meets the filtering labels.
+
+
+```yaml
+# This section configures the 'discovery service'
```

Comment:
If the discovery service is a product like the Teleport SSH/Database/Kubernetes etc. Service is, we should capitalize it (Discovery Service).